poc/ake_group.sage

```diff
@@ -0,0 +1,49 @@
+import sys
+import hashlib 
+
+########## Definitions from RFC 7748 ##################
+from sagelib.rfc7748 import *
+from sagelib.groups import *
+from sagelib.string_utils import *
+
+class GroupX25519(Group):
+    def __init__(self):
+        Group.__init__(self, "x25519")
+
+    def generator(self):
+        return IntegerToByteArray(9)
+
+    def serialize(self, element):
+        # Curve25519 points are bytes
+        return element
+
+    def deserialize(self, encoded):
+        # Curve25519 points are bytes
+        return encoded
+
+    def serialize_scalar(self, scalar):
+        # Curve25519 scalars are represented as bytes
+        return scalar
+
+    def element_byte_length(self):
+        return 32
+
+    def scalar_byte_length(self):
+        return 32
+
+    def random_scalar(self, rng):
+        return os.urandom(32)
```

Comment:
@chris-wood: This needs to apply the clamping operations to the scalar, otherwise it can generate an invalid X25519 private key

cc: @daxpedda

Reply:
Aha! Good catch.

Reply:
Let's revert to:

    key[0] &= 248
    key[31] &= 127
    key[31] |= 64
 
 as was originally suggested. Sorry for the confusion!